Frameworks/UIKit/UIApplicationMain.mm

```diff
@@ -384,8 +394,27 @@ void _UIApplicationShutdown() {
 }
 
 extern "C" void UIApplicationMainHandleProtocolEvent(IInspectable* protocolUri, const wchar_t* sourceApplication) {
-    WFUri* protocolResult = [WFUri createWith:protocolUri];
-    NSString* source = _bundleIdFromPackageFamilyName(sourceApplication);
+    id result = _createProjectionObject("WFUri", protocolUri, "protocol activation");
+
+    if (result) {
+        NSString* source = _bundleIdFromPackageFamilyName(sourceApplication);
+
+        [[UIApplication sharedApplication] _sendProtocolReceivedEvent:result source:source];
+    }
+}
+
+id _createProjectionObject(const char* className, IInspectable* source, const char* description) {
+    // We can't refer to projection classes directly, because we don't want UIKit to have any link-time
+    // dependencies on the projections. However, we can try to load the classes dynamically.
+
+    id result = nil;
+    id cls = objc_getClass(className);
+
+    if (cls) {
+        result = objc_msgSend(cls, @selector(createWith:), source);
```

Comment:
nit: `id objc_msgSend(id, SEL, ...)` does not guarantee that vararg calling convention matches that of fixed arguments. This requires a cast to be completely safe.

Alternatively,
```objc
@protocol _ProjectionClassInformal
+ (id)createWith:(IInspectable*)i;
@end
// ...
result = [static_cast<Class<_ProjectionClassInformal>>(cls) createWith:source]
```